File: engine/models/match-result.ts
import type { MatchEvent } from "./match-event";
import type { MatchState } from "./match-state";

/** Final result of a simulated match. */
export interface MatchResult {
  finalState: MatchState;
  events: readonly MatchEvent[];

  /** Player ID with the highest rating (Man of the Match). */
  mvpPlayerId: number;

  /** Player ID with the second highest rating. */
  svpPlayerId: number;
}

/** Convenience accessor for the home score. */
export function scoreHome(result: MatchResult): number {
  return result.finalState.scoreHome;
}

/** Convenience accessor for the away score. */
export function scoreAway(result: MatchResult): number {
  return result.finalState.scoreAway;
}

File: src/data/Models/MatchResult.cs
namespace ElevenLegends.Data.Models;

/// <summary>
/// Final result of a simulated match.
/// </summary>
public sealed record MatchResult
{
    public required MatchState FinalState { get; init; }
    public required IReadOnlyList<MatchEvent> Events { get; init; }

    /// <summary>Player ID with the highest rating (Man of the Match).</summary>
    public required int MvpPlayerId { get; init; }

    /// <summary>Player ID with the second highest rating.</summary>
    public required int SvpPlayerId { get; init; }

    public int ScoreHome => FinalState.ScoreHome;
    public int ScoreAway => FinalState.ScoreAway;
}
